File: samples/demo/Classes/SwitchTest/SwitchTest.cpp
#include "SwitchTest.h"

#define CAColor_blueStyle ccc4(51,204,255,255)

SwitchTest::SwitchTest()
{
	CADrawerController* drawer = (CADrawerController*)CAApplication::getApplication()->getRootWindow()->getRootViewController();
	drawer->setTouchMoved(false);
}

SwitchTest::~SwitchTest()
{
	CADrawerController* drawer = (CADrawerController*)CAApplication::getApplication()->getRootWindow()->getRootViewController();
	drawer->setTouchMoved(true);
}

void SwitchTest::viewDidLoad()
{
	size = this->getView()->getBounds().size;

	CASwitch* defaultSwitch = CASwitch::createWithCenter(CADipRect(size.width*0.5, size.height*0.2, size.width*0.3, 20));
	defaultSwitch->setTag(100);
	defaultSwitch->addTarget(this,CAControl_selector(SwitchTest::switchStateChange));
	this->getView()->addSubview(defaultSwitch);

	CASwitch* customSwitch = CASwitch::createWithCenter(CADipRect(size.width*0.5, size.height*0.4, size.width*0.3, 20));
	customSwitch->setTag(101);
	customSwitch->setOnImage(CAImage::create("source_material/btn_rounded_highlighted.png"));
	customSwitch->setOffImage(CAImage::create("source_material/btn_rounded_normal.png"));
	customSwitch->setThumbTintImage(CAImage::create("source_material/btn_rounded3D_selected.png"));
	customSwitch->addTarget(this, CAControl_selector(SwitchTest::switchStateChange));
	this->getView()->addSubview(customSwitch);

	switchState = CALabel::createWithCenter(CADipRect(size.width*0.5, size.height*0.6, size.width*0.5, 50));
	switchState->setColor(CAColor_blueStyle);
	switchState->setText("Switch State:Off");
	switchState->setFontSize(30 * CROSSAPP_ADPTATION_RATIO);
	switchState->setTextAlignment(CATextAlignmentCenter);
	switchState->setVerticalTextAlignmet(CAVerticalTextAlignmentCenter);
	this->getView()->addSubview(switchState);
}

void SwitchTest::switchStateChange(CAControl* btn, CCPoint point)
{
	CASwitch* state = (CASwitch*)btn;
	if (!state->isOn())
	{
		switchState->setText("Switch State:Off");
	}
	else
	{
		switchState->setText("Switch State:On");
	}
}

void SwitchTest::viewDidUnload()
{

}

File: samples/JavaTest/Resources/js/SwitchTest.js
var switchViewController =  ca.JSViewController.extend({
ctor:function () {
    this._super();
    var drawer = ca.Application.getApplication().getRootWindow().getRootViewController();
    drawer.setTouchMoved(false);
},
viewDidLoad:function(){
    var rect1 = this.getView().getBounds();
    var size = cc.size(rect1.width,rect1.height);
	var switchlabel  = ca.Label.createWithCenter(cc.rect(size.width/2,size.width/8,400,_px(80)));
	switchlabel.setText("Switch is off");
    switchlabel.setFontSize(_px(50));
    switchlabel.setTextAlignment(ca.TextAlignmentCenter);
    switchlabel.setColor(cc.c4b(0,0,0,255));
	switchlabel.setTag(500);
    this.getView().addSubview(switchlabel); 
                          
	var slider = ca.Slider.createWithCenter(cc.rect(size.width/2,size.height/2,300,80));
	slider.setTag(200);
    slider.addTarget(this,this.sliderCallBack);
    this.getView().addSubview(slider);
                                                       
    var label = ca.Label.createWithCenter(cc.rect(size.width/2,size.height*3/4,0.3125*size.height,size.width/8));
    label.setText("0.0");
    label.setTextAlignment(ca.TextAlignmentCenter);
    label.setFontSize(_px(50));
    label.setTag(300);
    label.setColor(cc.c4b(255,0,0,255));
    this.getView().addSubview(label);

	var progress = ca.Progress.create();
    progress.setTag(400);
    progress.setCenter(cc.rect(size.width/2,0.3125*size.height,400,200));
	this.getView().addSubview(progress);
	
    var switch1 = ca.Switch.createWithCenter(cc.rect(size.width/2,size.height/4,100,80));
    switch1.setTag(100);
    switch1.addTarget(this,this.switchCallBack);
    this.getView().addSubview(switch1);
                                                       
    var stepper = ca.Stepper.createWithCenter(cc.rect(size.width/2,0.8125*size.height,200,80));
    switch1.setTag(1000);
    this.getView().addSubview(stepper);
                                                       
},
viewDidUnload:function(){
    var drawer = ca.Application.getApplication().getRootWindow().getRootViewController();
    drawer.setTouchMoved(true);
},
sliderCallBack:function(sender,pos)
{
    var slider = sender;
    var value = slider.getValue();
    var label = this.getView().getSubviewByTag(300);
    label.setText(value);
    var progress = this.getView().getSubviewByTag(400);
    progress.setProgress(value);
                                                       
},
switchCallBack:function(sender,pos)
{
    var switchlabel = this.getView().getSubviewByTag(500);
    if(sender.isOn() == 0)
    {
    	switchlabel.setText("Switch is off");
    }else
    {
    	switchlabel.setText("Switch is on");
    }
    
},
updateCallBack:function(sender,pos)
{

}
});